doc/api/async_hooks.md

```diff
@@ -0,0 +1,275 @@
+# Async Hooks
+
+
+> Stability: 2 - Stable
+
+
+The `async-hooks` module provides an API to register callbacks tracking the lifetime of asynchronous handles created
+inside a Node.js application.
```

Comment:
Needs a terminology section: https://github.com/nodejs/node-eps/blob/e35d4813fdbbd99a751296e24361dba0d0dd9e10/XXX-asyncwrap-api.md#terminology

The top-level description should probably not include the word "handle".